### PR TITLE
158 pmsmart disable transfer of comments

### PR DESCRIPTION
**Pull request by ckunki**

### CODE REVIEW

File: logic/src/main/java/org/itsallcode/whiterabbit/logic/service/plugin/PluginWrapper.java [hunk 1]
```diff
@@ -93,10 +93,21 @@ public String toString()
 
     private class PluginConfigImpl implements PluginConfiguration
     {
+        private String prefixed(String key)
```
Comment (kaklakariada) [inline logic/src/main/java/org/itsallcode/whiterabbit/logic/service/plugin/PluginWrapper.java:96]: 👍

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/itsallcode/whiterabbit/api/PluginConfiguration.java [hunk 2]
```diff
@@ -18,4 +18,19 @@
...
+     * Get property {@code pluginId.propertyName} from the config file. If the
+     * property is not available then return defaultValue.
+     * 
+     * @param propertyName
+     *            the property name to get.
+     * @param defaultValue
+     *            alternative return value in case the property is not available
+     * @return the value of the property.
+     */
+    default boolean getOptionalValue(String propertyName, boolean defaultValue)
```
[the remark this answers is not in the document]
Comment (ckunki) [reply api/src/main/java/org/itsallcode/whiterabbit/api/PluginConfiguration.java:32]: I thought my idea was clever. 
But looking at the code after applying your proposal I like it better. 
Thanks.

File: logic/src/test/java/org/itsallcode/whiterabbit/logic/ConfigFileTest.java [hunk 3]
```diff
@@ -104,4 +106,18 @@ void getLocale_returnsCustomValue()
         when(propertiesMock.getProperty("locale")).thenReturn("zh");
         assertThat(configFile.getLocale()).isEqualTo(Locale.forLanguageTag("zh"));
     }
+
+    @Test
+    void mandatoryValue()
```
Comment (kaklakariada) [inline logic/src/test/java/org/itsallcode/whiterabbit/logic/ConfigFileTest.java:111]:
```suggestion
    void mandatoryValueFailsForMissingValue()
```

File: logic/src/test/java/org/itsallcode/whiterabbit/logic/ConfigFileTest.java [hunk 4]
```diff
@@ -104,4 +106,18 @@ void getLocale_returnsCustomValue()
...
+    {
+        assertThrows(IllegalStateException.class, () -> {
+            configFile.getMandatoryValue("missing.mandatory.property");
+        });
+    }
+
+    @Test
+    void optionalValue()
```
Comment (kaklakariada) [inline logic/src/test/java/org/itsallcode/whiterabbit/logic/ConfigFileTest.java:119]:
```suggestion
    void optionalValueReturnsEmptyOptionalForMissingValue()
```

File: logic/src/test/java/org/itsallcode/whiterabbit/logic/ConfigFileTest.java [hunk 5]
```diff
@@ -104,4 +106,18 @@ void getLocale_returnsCustomValue()
...
+    {
+        assertThat(configFile.getOptionalValue("missing.optional.property")).isEqualTo(Optional.empty());
```
Comment (kaklakariada) [inline logic/src/test/java/org/itsallcode/whiterabbit/logic/ConfigFileTest.java:121]:
```suggestion
        assertThat(configFile.getOptionalValue("missing.optional.property")).isEmpty();
```

File: plugins/pmsmart/src/main/java/org/itsallcode/whiterabbit/plugin/pmsmart/PMSmartExporter.java [hunk 6]
```diff
@@ -44,7 +44,10 @@ public void export(ProjectReport report, ProgressMonitor progressMonitor)
         try (final var driver = webDriverFactory.createWebDriver(baseUrl))
         {
             final var weekViewPage = driver.getWeekViewPage();
-            new ExportHelper(progressMonitor, weekViewPage).export(daysToExport);
+            new ExportHelper(progressMonitor, weekViewPage)
+                    .withTransferComments(
+                            Boolean.parseBoolean(config.getOptionalValue("transfer.comments").orElse("true")))
```
Comment (kaklakariada) [inline plugins/pmsmart/src/main/java/org/itsallcode/whiterabbit/plugin/pmsmart/PMSmartExporter.java:49]:
```suggestion
                            config.getOptionalValue("transfer.comments").map(Boolean::parseBoolean).orElse(true))
```
I didn't try this code, please test before using it ;)

File: plugins/pmsmart/src/test/java/org/itsallcode/whiterabbit/plugin/pmsmart/PMSmartExporterTest.java [hunk 7]
```diff
@@ -71,6 +72,26 @@ void exportActivity()
...
 
+    @Test
+    void exportActivityWithoutComments()
+    {
+        when(configMock.getOptionalValue("transfer.comments")).thenReturn(Optional.of("false"));
+
+        final LocalDate date = LocalDate.of(2021, Month.MAY, 3);
+        runExport(day(date, DayType.WORK, activity(COST_CARRIER, Duration.ofHours(1), "project1")));
+        verify(projectRowMock).enterDuration(date, Duration.ofHours(1));
+        verify(projectRowMock, never()).enterComment(null, "never");
```
Comment (kaklakariada) [inline plugins/pmsmart/src/test/java/org/itsallcode/whiterabbit/plugin/pmsmart/PMSmartExporterTest.java:83]:
```suggestion
        verify(projectRowMock, never()).enterComment(any(), any());
```
When using never(), better be more generic with the arguments to be sure the method is not called at all.
You might need to static import `Mockito.any()`

File: plugins/pmsmart/src/test/java/org/itsallcode/whiterabbit/plugin/pmsmart/PMSmartExporterTest.java [hunk 8]
```diff
@@ -71,6 +72,26 @@ void exportActivity()
...
+    }
+
+    @Test
+    void exportActivityWithCommentsSetToTrue()
```
Comment (kaklakariada) [inline plugins/pmsmart/src/test/java/org/itsallcode/whiterabbit/plugin/pmsmart/PMSmartExporterTest.java:87]: 👍

**Review by kaklakariada — CHANGES_REQUESTED** [review]
👍 Only minor findings :)

**Comment by kaklakariada** [comment]
Please also add a changelog entry.

File: plugins/pmsmart/src/main/java/org/itsallcode/whiterabbit/plugin/pmsmart/PMSmartExporter.java (hunk 6 above)
Comment (ckunki) [reply plugins/pmsmart/src/main/java/org/itsallcode/whiterabbit/plugin/pmsmart/PMSmartExporter.java]: Yes, I will try, 
But at first glance: Much better, thanks!

File: CHANGELOG.md [hunk 9]
```diff
@@ -8,6 +8,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 See [Release](https://github.com/itsallcode/white-rabbit/releases/tag/v1.7.0) / [Milestone](https://github.com/itsallcode/white-rabbit/milestone/9?closed=1)
 
+* [#149](https://github.com/itsallcode/white-rabbit/issues/149): Enlarged width/height of scrollbars to make them easier to click on.
+* [#152](https://github.com/itsallcode/white-rabbit/issues/152): Display short day of week without dot or comma.
+* [#111](https://github.com/itsallcode/white-rabbit/issues/111): Display week ends in other color.
+* [#151](https://github.com/itsallcode/white-rabbit/issues/151): Fixed failure in PM Smart export if current month view showed a day number and the same number being used as the number of a calendar week.
```
Comment (kaklakariada) [inline CHANGELOG.md:14]:
```suggestion
```
These changes are already contained in version 1.6.0

**Review by kaklakariada — CHANGES_REQUESTED** [review]
Only minor change in changelog

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**